src/connector/src/source/google_pubsub/split.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::anyhow;
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+
+use crate::source::{SplitId, SplitMetaData};
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Hash)]
+pub struct PubsubSplit {
+    pub(crate) index: u32,
+    pub(crate) subscription: String,
+    pub(crate) start_offset: Option<String>,
```

Comment:
what is the format? a numeric timestamp or like yyyy-MM-dd?
I hope users can set the start offset manually

Reply:
Numeric timestamp. Will document